rclpy/rclpy/node.py

```diff
@@ -715,9 +715,11 @@ def _set_parameters_atomically(
         if not result.successful:
             return result
         elif self._parameters_callback:
-            result = self._parameters_callback(parameter_list)
-        else:
-            result = SetParametersResult(successful=True)
+            for callback in self._parameters_callback:
+                result = callback(parameter_list)
```

Comment:
how about 

```
result = callback(parameter_list) and result
```

instead? Otherwise, any failed callback functions will stop the execution of the rest.

Reply:
@allenh1 sir bur here it is mentioned that thing ros2/rclcpp#722.Also I looked into @wjwwood c++ code and he also stopped the execution when a unsuccessful result is throw.

Reply:
> Also I looked into @wjwwood c++ code and he also stopped the execution when a unsuccessful result is throw.

That's fine then. It should mimic the C++ implementation.